Source/Core/Core/IOS/FS/FileSystem.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/IOS/FS/FileSystem.h"
+
+#include "Common/FileUtil.h"
+#include "Core/IOS/FS/HostBackend/FS.h"
+
+namespace IOS::HLE::FS
+{
+std::unique_ptr<FileSystem> MakeFileSystem(bool use_session_path)
```

Comment:
This should probably be using an enum class. Think of a potential call site like:

```cpp
... = MakeFileSystem(false);
```

From the perspective of someone who isn't acquainted with the API, they'd wonder "what does this do"? Currently it's ambiguous. It's communicating *not* to make the filesystem but returns a live pointer to a filesystem that was, in fact, created.

Reply:
Hmm, that's true. I've replaced it with an enum class parameter.